seed/csharp-sdk/alias-extends/src/SeedAliasExtends.Test/Core/RawClientTests/AdditionalParametersTests.cs

```diff
@@ -0,0 +1,300 @@
+                {
+                    new("foo", "bar"),
+                },
+            },
+        };
+
+        var response = await _rawClient.SendRequestAsync(request);
+        Assert.That(response.StatusCode, Is.EqualTo(200));
+
+        var content = await response.Raw.Content.ReadAsStringAsync();
+        Assert.That(content, Is.EqualTo("Success"));
+
+        Assert.That(_server.LogEntries.Count, Is.EqualTo(1));
+    }
+
+    [Test]
+    public async SystemTask SendRequestAsync_AdditionalQueryParameters_Override()
+    {
+        _server
+            .Given(WireMockRequest.Create().WithPath("/test").WithParam("foo", "null").UsingGet())
+            .RespondWith(WireMockResponse.Create().WithStatusCode(200).WithBody("Success"));
+
+        var request = new RawClient.JsonApiRequest()
+        {
+            BaseUrl = _baseUrl,
+            Method = HttpMethod.Get,
+            Path = "/test",
+            Query = new Dictionary<string, object> { { "foo", "bar" } },
+            Options = new RequestOptions
+            {
+                AdditionalQueryParameters = new List<KeyValuePair<string, string>>
+                {
+                    new("foo", "null"),
+                },
+            },
+        };
+
+        var response = await _rawClient.SendRequestAsync(request);
+        Assert.That(response.StatusCode, Is.EqualTo(200));
+
+        var content = await response.Raw.Content.ReadAsStringAsync();
+        Assert.That(content, Is.EqualTo("Success"));
+
+        Assert.That(_server.LogEntries.Count, Is.EqualTo(1));
+    }
+
+    [Test]
+    public async Task SendRequestAsync_AdditionalQueryParameters_Merge()
+    {
+        // WireMock server expectation with all three 'foo' parameters
+        _server
+            .Given(WireMockRequest.Create().WithPath("/test").UsingGet())
+            .RespondWith(WireMockResponse.Create().WithStatusCode(200).WithBody("Success"));
+
+        var request = new RawClient.JsonApiRequest()
+        {
+            BaseUrl = _baseUrl,
+            Method = HttpMethod.Get,
+            Path = "/test",
+            Query = new Dictionary<string, object> { { "foo", "baz" } },
+            Options = new RequestOptions
+            {
+                AdditionalQueryParameters = new List<KeyValuePair<string, string>>
+                {
+                    new("foo", "one"),
+                    new("foo", "two"),
+                },
+            },
+        };
+
+        var response = await _rawClient.SendRequestAsync(request);
+        Assert.That(response.StatusCode, Is.EqualTo(200));
+
+        var content = await response.Raw.Content.ReadAsStringAsync();
+        Assert.That(content, Is.EqualTo("Success"));
+
+        Assert.That(_server.LogEntries.Count, Is.EqualTo(1));
+
+        var requestUrl = _server.LogEntries.First().RequestMessage.Url;
+        Assert.That(requestUrl, Does.Contain("foo=one"));
+        Assert.That(requestUrl, Does.Contain("foo=two"));
```

Comment:
This does not verify whether `foo=baz` is removed

Reply:
Good call - I'll add that in.